mcs_new_builds/analysis.py

```diff
@@ -0,0 +1,195 @@
+# %% [markdown]
+# ### Imports and setup
+
+# %%
+from asf_core_data import get_mcs_installations, load_preprocessed_epc_data
+
+import pandas as pd
+import numpy as np
+
+# %%
+# change path to your own local version of EPC data
+epc_path = "/Users/chris.williamson/Documents/ASF_data"
+
+# %%
+mcs = get_mcs_installations("full")
+
+# %%
+# convert date columns to datetime type
+mcs["commission_date"] = pd.to_datetime(mcs["commission_date"])
+mcs["INSPECTION_DATE"] = pd.to_datetime(mcs["INSPECTION_DATE"])
+
+# %%
+# merge installation type columns and filter to domestic installations
+mcs["installation_type"] = mcs["installation_type"].fillna(
+    mcs["end_user_installation_type"]
+)
+mcs = mcs.loc[mcs.installation_type == "Domestic"].reset_index(
+    drop=True
+)
+
+# %% [markdown]
+# ### What proportion of domestic records in the MCS database relate to new-build installations?
+
+# %%
+# UPRNs of properties that have an EPC labelled as "new dwelling" (not necessarily the first one - see below)
+new_uprns = [uprn for uprn in mcs.loc[mcs["TRANSACTION_TYPE"] == "new dwelling"]["UPRN"] if uprn != "unknown"]
+
+# %%
+# filter to first records
+# records not linked to an EPC are kept
+first_records = (
+    mcs
+    .sort_values("INSPECTION_DATE")
+    .groupby("original_mcs_index")
+    .head(1)
+    .sort_values("original_mcs_index")
+)
+
+# %%
+# find number of days between first recorded EPC inspection and HP commission
+# could instead use the first EPC that labels the property as a "new dwelling", rather than the first overall - which is best? (see below)
+first_records["diff_epc_to_mcs"] = (
+    first_records["commission_date"] - first_records["INSPECTION_DATE"]
+).dt.days
```

Comment:
I'm not sure it matters too much - my thinking here was that most properties would have commissioning after inspection, and the ones we're looking for would be unusual since this value would generally be negative. But this would all work the same with the signs flipped